src/content/1.7/development/components/grid/tutorials/work-with-search-form/_index.md

```diff
@@ -0,0 +1,293 @@
+            $request,
+            ManufacturerGridDefinitionFactory::GRID_ID,
+            'admin_manufacturers_index'
+        );
+    }
+    ...
+}
+```
+
+```yaml
+# your-module/config/routes.yml
+admin_manufacturers_search:
+  path: /
+  methods: POST
+  defaults:
+    _controller: 'PrestaShopBundle:Admin/Sell/Catalog/Manufacturer:search'
+    _legacy_controller: AdminManufacturers
+    _legacy_link: AdminManufacturers:submitFiltermanufacturer
+```
+
+#### Common search controller
+{{< minver v="1.7.7" title="true" >}}
+
+As this controller is almost always the same, starting 1.7.7, we introduced a common controller. So all you need to do is define the routing:
+
+```yaml
+# your-module/config/routes.yml
+admin_addresses_search:
+  path: /
+  methods: [POST]
+  defaults:
+    _controller: PrestaShopBundle:Admin\Common:searchGrid
+    gridDefinitionFactoryServiceId: prestashop.core.grid.definition.factory.address
+    redirectRoute: admin_addresses_index
+    _legacy_controller: AdminAddresses
+    _legacy_link: AdminAddresses:submitFilteraddress
+```
+
+### List action
+
+Thanks to the internal parameter resolver you can directly use your `Filters` class as an argument in the controller. It then automatically:
+
+- parses the potential parameters in the query and fills them into the `Filters` argument
+- matches thanks to the class the filters from the database and fetch them if present
+- persists the filters in the database
+
+```php
+class ManufacturerController extends FrameworkBundleAdminController
+{
+    ...
+    /**
+     * Show manufacturers listing page.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     * @param ManufacturerFilters $manufacturerFilters
+     *
+     * @return Response
+     */
+    public function indexAction(
+        Request $request,
+        ManufacturerFilters $manufacturerFilters
+    ) {
+        $manufacturerGridFactory = $this->get('prestashop.core.grid.grid_factory.manufacturer');
+        $manufacturerGrid = $manufacturerGridFactory->getGrid($manufacturerFilters);
+
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Manufacturer/index.html.twig', [
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'manufacturerGrid' => $this->presentGrid($manufacturerGrid),
+        ]);
+    }
+    ...
+}
+```
+
+### Reset action
+
+Last action is used to reset the persisted filters and your grid filtering/sorting. This action is the same for nearly all grids so PrestaShop provides a common controller to manage it, and you actually already set it via the grid definition.
+
+It is defined in the `SearchAndResetType` options, it uses `admin_common_reset_search_by_filter_id` and only needs the filter id to identify the filters to clear, and a redirection route.
```

Comment:
For `SearchAndResetType`, add a link to the first file. Improve the internal linking.